core/src/main/java/me/lokka30/treasury/plugin/core/schedule/Scheduler.java

```diff
@@ -26,4 +29,130 @@ public interface Scheduler {
      */
     void runAsync(Runnable task);
 
+    default Scheduled runAsync(
+            Runnable task, long delay, long repeat, TimeUnit unit
+    ) {
+        Scheduled scheduled = new Scheduled() {
+
+            private AtomicBoolean cancelled = new AtomicBoolean(false);
+            private AtomicBoolean done = new AtomicBoolean(false);
+
+            private long delayInner = delay;
+            private long repeatInner = repeat;
+            private TimeUnit unitInner = unit;
+
+            @Override
+            public void start(long delayNew, long repeatNew, TimeUnit unitNew) {
+                if (cancelled.get()) {
+                    cancelled.set(false);
+                }
+                if (done.get()) {
+                    done.set(false);
+                }
+                this.delayInner = delayNew;
+                this.repeatInner = repeatNew;
+                this.unitInner = unitNew;
+                this.run();
+            }
+
+            @Override
+            public boolean cancelled() {
+                return cancelled.get();
+            }
+
+            @Override
+            public boolean done() {
+                return done.get() || cancelled.get();
+            }
+
+            @Override
+            public void cancel() {
+                cancelled.set(true);
+            }
+
+            @Override
+            public void run() {
+                if (cancelled.get()) {
+                    return;
+                }
+                runAsync(() -> {
+                    if (delayInner > 0) {
+                        try {
+                            Thread.sleep(unitInner.toMillis(delayInner));
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                        }
+                    }
+                    while (!cancelled.get()) {
+                        if (cancelled.get()) {
+                            break;
+                        }
```

Comment:
This condition is unnecessary here, if the `cancelled#get` returns true the while loop will exits the operation.